portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,32 +25,51 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.PreparedQuery;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+
   private static final Gson gson = new Gson();  
-  private static final ArrayList<String> comments = new ArrayList<String>();  
+  private static final ArrayList<String> commentList = new ArrayList<String>();
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+  private static final String COMMENT = "Comment";  
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query(COMMENT);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty(COMMENT);
+        commentList.add(comment);
```

Comment:
`commentList` is never cleared, so it's always being added to. Try this in your dev server. Add a comment or 2, then retrieve comments through your `doGet` call. See how the comments keep stacking on top of each other. This is because every time you call `doGet`, all the comments in your database are being added to `commentList`.

Before, we had `commentList` as your "in-memory" database. Now that you have a legitimate database, you don't need to have the `commentList` field anymore. It can just be a local variable in `doGet`.

Reply:
Ok so the commentList was basically mirroring the datastore and that's unnecessary. I understand that.